processing/src/main/java/org/apache/druid/frame/segment/FrameSegment.java

```diff
@@ -77,4 +80,15 @@ public void close()
   {
     // Nothing to close.
   }
+
+  @SuppressWarnings("unchecked")
+  @Nullable
+  @Override
+  public <T> T as(@Nonnull Class<T> clazz)
+  {
+    if (CloseableShapeshifter.class.equals(clazz)) {
```

Comment:
This code is breaking a code path where default method is called with StorageAdaptor.class

```
 default <T> T as(@Nonnull Class<T> clazz)
  {
    if (clazz.equals(QueryableIndex.class)) {
      return (T) asQueryableIndex();
    } else if (clazz.equals(StorageAdapter.class)) {
      return (T) asStorageAdapter();
    }
    return null;
  }
}
```
Is that okay ?

Reply:
Should it be `else return super.as(clazz)` instead of `else return null`?

Reply:
+1

Reply:
I looked at the comment in 
```
* Implementations which accept classes other than {@link QueryableIndex} or {@link StorageAdapter} are limited 
   * to using those classes within the extension. This means that one extension cannot rely on the `Segment.as` 
   * behavior of another extension.
 ```
 Since here we are using a different one here, based on the comment I resorted to returning null. The same pattern is observed in https://github.com/apache/druid/blob/50bae96e8b44913faeb7c64a79430ffd7cb843a4/processing/src/main/java/org/apache/druid/segment/QueryableIndexSegment.java#L78. Can you please help me understand better @gianm  @cryptoe